src/annotate.ts

```diff
@@ -43,43 +43,38 @@ export default async function annotate({payload, _ajv = ajv, _schema = schema}:
     json: true
   };
 
-  let json: any = await _request(options)
+  const json: any = await _request(options)
     .then((res: any) => {
       return res;
     })
     .catch((err: any) => {
-      //console.error('_request error: ' + err);
       throw new Error('_request error: ' + err);
     });
 
-  json = json[0];
-  //console.log(JSON.stringify(json, undefined, 2));
-  // if there is no match, returns undefined
-  // if there are multiple matches, returns only the first one.. check this is appropriate.
+  // if there is no match, the request returns [undefined]
+  // if there are multiple matches, returns all.
 
-  if (json) {
+  if (json.length > 0 && typeof json[0] !== 'undefined') {
+    return json.map((x: any) => {
+      const uriSplit = x._links.olslinks[0].semanticTag.split('/');
 
-    const uriSplit = json._links.olslinks[0].semanticTag.split('/');
+      // make ontologyShortName equal to ontology accession prefix
+      let short_name = uriSplit.slice(-1)[0].split('_')[0].toLowerCase();
+      if (['format', 'operation', 'topic'].indexOf(short_name) > -1) short_name = 'edam';
+      if (short_name === 'orphanet') short_name = 'ordo';
 
-    // make ontologyShortName equal to ontology accession prefix
-    let ontologyShortName = uriSplit.slice(-1)[0].split('_')[0].toLowerCase();
-    if (['format', 'operation', 'topic'].indexOf(ontologyShortName) > -1) ontologyShortName = 'edam';
-    if (ontologyShortName === 'orphanet') ontologyShortName = 'ordo';
+      return {
+        term: x.annotatedProperty.propertyValue.toLowerCase(),
+        iri: x._links.olslinks[0].semanticTag,
+        confidence: x.confidence,
```

Comment:
We are still returning confidence as a `string`. Is this the expected behaviour for this PR?